### PR TITLE
fix(kit): handle tuple-format modules in `hasNuxtModule`

### PR DESCRIPTION
**Pull request by harlan-zw**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The `hasNuxtModule` function is incorrectly returning false for the following conditions:

- module with inline options
- module instance (with and without inline options)

```ts
export default defineNuxtConfig({
  modules: [
    ['my-module', { foo: 'bar' }],
   defineNuxtModule({ name: 'my-module' }),
   [defineNuxtModule({ name: 'my-module' }), { foo: 'bar' }],
  ]
})
```

Additionally, `getNuxtModuleVersion` is failing when using any of the above.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/kit/src/module/compatibility.ts [hunk 1]
```diff
@@ -11,8 +11,22 @@ import { loadNuxtModuleInstance } from './install'
...
+              // input may either a string, an array or a module instance
+              function resolveModuleEntry (input: typeof m): boolean {
+                if (typeof input === 'object' && !Array.isArray(input)) {
+                  return (input as any as NuxtModule).name === moduleName
+                }
+                return Array.isArray(input) ? resolveModuleEntry(input[0]) : input === moduleName
+              }
+              return resolveModuleEntry(m)
+            })
+        )
```
Comment (danielroe) [inline packages/kit/src/module/compatibility.ts:29]: Maybe we could use `modules.some`?

Comment (harlan-zw) [reply packages/kit/src/module/compatibility.ts]: Always forget about some, thanks!

File: packages/kit/src/module/compatibility.ts [hunk 2]
```diff
@@ -11,8 +11,20 @@ import { loadNuxtModuleInstance } from './install'
...
  */
 export function hasNuxtModule (moduleName: string, nuxt: Nuxt = useNuxt()) : boolean {
+  // check installed modules
   return nuxt.options._installedModules.some(({ meta }) => meta.name === moduleName) ||
-    nuxt.options.modules.includes(moduleName)
+    // check modules to be installed
+    nuxt.options.modules
+      .some((m) => {
+        // input may either a string, an array or a module instance
+        function resolveModuleEntry (input: typeof m): boolean {
```
Comment (danielroe) [inline packages/kit/src/module/compatibility.ts:20]: Do you think we can hoist this out of the inline function?

Comment (harlan-zw) [reply packages/kit/src/module/compatibility.ts]: Good idea :+1:

File: packages/kit/src/module/compatibility.ts [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
  *
  * This will check both the installed modules and the modules to be installed. Note
  * that it cannot detect if a module is _going to be_ installed programmatically by another module.
  */
 export function hasNuxtModule (moduleName: string, nuxt: Nuxt = useNuxt()) : boolean {
+  // check installed modules
   return nuxt.options._installedModules.some(({ meta }) => meta.name === moduleName) ||
-    nuxt.options.modules.includes(moduleName)
+    // check modules to be installed
+    nuxt.options.modules.some(m => resolveNuxtModuleEntryName(m))
```
Comment (danielroe) [inline packages/kit/src/module/compatibility.ts:27]:
```suggestion
    nuxt.options.modules.some(m => moduleName === resolveNuxtModuleEntryName(m))
```

File: packages/kit/src/module/compatibility.ts [hunk 4]
```diff
@@ -1,18 +1,30 @@
 import satisfies from 'semver/functions/satisfies.js' // npm/node-semver#381
-import type { Nuxt, NuxtModule } from '@nuxt/schema'
+import type { Nuxt, NuxtModule, NuxtOptions } from '@nuxt/schema'
 import { useNuxt } from '../context'
 import { normalizeSemanticVersion } from '../compatibility'
 import { loadNuxtModuleInstance } from './install'
 
+export function resolveNuxtModuleEntryName (m: NuxtOptions['modules'][number]): string | false {
```
Comment (danielroe) [inline packages/kit/src/module/compatibility.ts:7]:
```suggestion
function resolveNuxtModuleEntryName (m: NuxtOptions['modules'][number]): string | false {
```

**Review by danielroe — APPROVED** [review]

**Pull request merged**